File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        
        if (arr.length == 1) {
            return new int[] {-1};
        }
        
        int[] answer = new int[arr.length - 1];
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < min) {
                min = arr[i];
            }
        }
        
        int index = 0;
        for (int num : arr) {
            if (num == min) continue;
            answer[index++] = num;
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> arr) {
    
    if (arr.size() == 1) return {-1};
    
    
    vector<int> answer;
    int min = *min_element(arr.begin(), arr.end());
    
    for (int num : arr) {
        if (min != num) answer.push_back(num);
    }
    
    return answer;
}